reconcile/test/test_vault_replication.py

```diff
@@ -0,0 +1,182 @@
+from typing import cast
+
+import pytest
+
+import reconcile.vault_replication as integ
+from reconcile.gql_definitions.fragments.vault_secret import VaultSecret
+from reconcile.gql_definitions.jenkins_configs.jenkins_configs import (
+    JenkinsConfigsQueryData,
+    JenkinsConfigV1_JenkinsConfigV1,
+    JenkinsInstanceV1,
+    ResourceV1,
+)
+from reconcile.gql_definitions.vault_instances.vault_instances import (
+    VaultReplicationConfigV1_VaultInstanceAuthV1,
+    VaultReplicationConfigV1_VaultInstanceAuthV1_VaultInstanceAuthApproleV1,
+)
+from reconcile.gql_definitions.vault_policies import vault_policies
+from reconcile.utils.vault import (
+    VaultClient,
+    _VaultClient,
+)
+
+
+def test_policy_contais_path():
+    policy_paths = ["path1", "path2"]
+    path = "path1"
+    assert integ.policy_contains_path(path, policy_paths) is True
+
+
+def test_policy_contais_path_false():
+    policy_paths = ["path2", "path3"]
+    path = "path1"
+    assert integ.policy_contains_path(path, policy_paths) is False
+
+
+def test_check_invalid_paths_ko():
+    path_list = ["path1", "path3"]
+    policy_paths = ["path1", "path2"]
+    with pytest.raises(integ.VaultInvalidPaths):
+        integ.check_invalid_paths(path_list, policy_paths)
+
+
+def test_check_invalid_paths_ok():
+    path_list = ["path1", "path2"]
+    policy_paths = ["path1", "path2"]
+    integ.check_invalid_paths(path_list, policy_paths)
+
+
+def test_list_invalid_paths():
+    path_list = ["path1", "path3"]
+    policy_paths = ["path1", "path2"]
+    assert integ.list_invalid_paths(path_list, policy_paths) == ["path3"]
+
+
+@pytest.fixture
```

Comment:
nit: usually we declare all fixtures at the beginning of a test file and then we write all the tests

Reply:
Changed!